test/unit/firebase-app.spec.ts

```diff
@@ -149,6 +153,64 @@ describe('FirebaseApp', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config.json';
+      const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
+      delete process.env[FIREBASE_CONFIG_FILE_VAR];
+      expect(app.options.databaseURL).to.equal("https://hipster-chat.firebaseio.mock");
+      expect(app.options.projectId).to.equal("hipster-chat-mock");
+      expect(app.options.storageBucket).to.equal("hipster-chat.appspot.mock");
+    });
+
+    it('should complain about a non existant file', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_non_existant.json';
+      expect(() => {
+       const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
+      }).to.throw(`Failed to read app options file: Error: ENOENT: no such file or directory, open './test/resources/firebase_config_non_existant.json'`);
+      delete process.env[FIREBASE_CONFIG_FILE_VAR];
+    });
+
+    it('should complain about a non parsable file', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_bad.json';
+      expect(() => {
+        const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
+      }).to.throw(`Failed to parse app options file: SyntaxError: Unexpected end of JSON input`);
+      delete process.env[FIREBASE_CONFIG_FILE_VAR];
+    });
+
+    it('should use the existing values when some of them exist', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config.json';
+      const app = firebaseNamespace.initializeApp(mocks.appOptions, mocks.appName);
+      delete process.env[FIREBASE_CONFIG_FILE_VAR];
+      expect(app.options.databaseURL).to.equal('https://databaseName.firebaseio.com');
+      expect(app.options.projectId).to.equal('hipster-chat-mock');
+      expect(app.options.storageBucket).to.equal('bucketName.appspot.com');
+    });
+
+    it('should use the existing values when some of them aren\'t overwritten', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_partial.json';
+      const app = firebaseNamespace.initializeApp(mocks.appOptionsAuthDB, mocks.appName);
+      delete process.env[FIREBASE_CONFIG_FILE_VAR];
+      expect(app.options.databaseURL).to.equal('https://databaseName.firebaseio.com');
+      expect(app.options.projectId).to.equal('hipster-chat-mock');
+      expect(app.options.storageBucket).to.equal(undefined);
+    });
+
+    it('should init without problem when the config env var isn\'t set', () => {
+      const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
+      expect(app.options.databaseURL).to.equal(undefined);
+      expect(app.options.projectId).to.equal(undefined);
+      expect(app.options.storageBucket).to.equal(undefined);
+    });
+
+    it('should init with no params setting default credential', () => {
```

Comment:
Add another test case where the environment variable is set.

Reply:
Ok, it wasn't clear what's going on here. Shall we update the description to something like 'should init with application default credentials when no options provided and the environment variable is not set'?